metricsaggregator/src/main/java/io/dockstore/metricsaggregator/helper/AggregationHelper.java

```diff
@@ -249,12 +258,71 @@ public static Optional<MemoryMetric> getAggregatedMemoryFromExecutions(List<RunE
+            DoubleStatistics statistics = new DoubleStatistics(memoryRequirements);
             return Optional.of(new MemoryMetric()
-                    .minimum(statistics.min())
-                    .maximum(statistics.max())
-                    .average(statistics.average())
-                    .numberOfDataPointsForAverage(statistics.numberOfDataPoints()));
+                    .minimum(statistics.getMinimum())
+                    .maximum(statistics.getMaximum())
+                    .average(statistics.getAverage())
+                    .numberOfDataPointsForAverage(statistics.getNumberOfDataPoints()));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Aggregate Cost metrics from all submissions by calculating the minimum, maximum, and average.
+     * @param allSubmissions
+     * @return
+     */
+    public static Optional<CostMetric> getAggregatedCost(ExecutionsRequestBody allSubmissions) {
+        // Get aggregated cost metrics that were submitted to Dockstore
+        List<CostMetric> costMetrics = allSubmissions.getAggregatedExecutions().stream()
+                .map(Metrics::getCost)
+                .filter(Objects::nonNull)
+                .collect(Collectors.toCollection(ArrayList::new));
+        getAggregatedCostFromExecutions(allSubmissions.getRunExecutions()).ifPresent(costMetrics::add);
+
+        if (!costMetrics.isEmpty()) {
+            List<MoneyStatistics> statistics = costMetrics.stream()
+                    .map(metric -> new MoneyStatistics(Money.of(metric.getMinimum(), metric.getUnit()), Money.of(metric.getMaximum(), metric.getUnit()), Money.of(metric.getAverage(),
+                            metric.getUnit()), metric.getNumberOfDataPointsForAverage()))
+                    .toList();
+            MoneyStatistics moneyStatistics = MoneyStatistics.createFromStatistics(statistics);
+            return Optional.of(new CostMetric()
+                    .minimum(moneyStatistics.getMinimum().getNumberStripped().doubleValue())
+                    .maximum(moneyStatistics.getMaximum().getNumberStripped().doubleValue())
+                    .average(moneyStatistics.getAverage().getNumberStripped().doubleValue())
+                    .numberOfDataPointsForAverage(moneyStatistics.getNumberOfDataPoints()));
+        }
+        return Optional.empty();
+    }
+
+    /**
+     * Aggregate Cost metrics from the list of run executions by calculating the minimum, maximum, and average.
+     * @param executions
+     * @return
+     */
+    public static Optional<CostMetric> getAggregatedCostFromExecutions(List<RunExecution> executions) {
+        List<Cost> submittedCosts = executions.stream()
+                .map(RunExecution::getCost)
+                .filter(Objects::nonNull)
+                .toList();
+
+        boolean containsMalformedCurrencies = submittedCosts.stream().anyMatch(cost -> !isValidCurrencyCode(cost.getCurrency()));
```

Comment:
The Java Money library can [convert currencies](https://www.baeldung.com/java-money-and-currency#currency-conversion), so I think we would convert it all to one currency (maybe USD) then aggregate

Reply:
Interesting, I did not know that. 

I don't know finance, but I wonder if we should track when we calculate the exchange? For example, a currency is devalued; if we calculate the exchange the day before the devaluation, it evaluates to $10, but if we calculate it the day after the devaluation, it comes to $6...

But I don't know finance, and we're getting into real hypotheticals here; I don't think this needs to block the PR.

Reply:
That's a good point and something I did ponder when deciding if the endpoint should allow the submission of any currency in the first iteration of the cost metric. Ultimately decided to keep things simple by only allowing USD submissions, so we don't need to worry about things like that

This does make me wonder if we should record the date that the workflow was executed because a workflow run that costs $5 in 2020 might not cost $5 today :thinking: